main/ui/src/main/java/org/cryptomator/ui/stats/VaultStatisticsController.java

```diff
@@ -0,0 +1,213 @@
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.ResourceBundle;
+
+@VaultStatisticsScoped
+public class VaultStatisticsController implements FxController {
+
+	private static final int IO_SAMPLING_STEPS = 30;
+	private static final double IO_SAMPLING_INTERVAL = 1;
+
+	private final VaultStats stats;
+	private final Series<Number, Number> readData;
+	private final Series<Number, Number> writeData;
+	private final Timeline ioAnimation;
+	private final LongBinding bpsRead;
+	private final LongBinding bpsWritten;
+	private final DoubleBinding cacheHitRate;
+	private final DoubleBinding cacheHitDregrees;
+	private final DoubleBinding cacheHitPercentage;
+	private final LongBinding totalBytesRead;
+	private final LongBinding totalBytesWritten;
+	private final LongBinding totalBytesEncrypted;
+	private final LongBinding totalBytesDecrypted;
+	private final LongBinding filesRead;
+	private final LongBinding filesWritten;
+	private final LongBinding bpsEncrypted;
+	private final LongBinding bpsDecrypted;
+
+	public AreaChart<Number, Number> readChart;
+	public AreaChart<Number, Number> writeChart;
+	public NumberAxis readChartXAxis;
+	public NumberAxis readChartYAxis;
+	public NumberAxis writeChartXAxis;
+	public NumberAxis writeChartYAxis;
+
+	@Inject
+	public VaultStatisticsController(@VaultStatisticsWindow Stage window, @VaultStatisticsWindow Vault vault, ResourceBundle resourceBundle) {
+		this.stats = vault.getStats();
+		this.readData = new Series<>();
+		this.writeData = new Series<>();
+		this.bpsRead = WeakBindings.bindLong(stats.bytesPerSecondReadProperty());
+		this.bpsWritten = WeakBindings.bindLong(stats.bytesPerSecondWrittenProperty());
+		this.cacheHitRate = WeakBindings.bindDouble(stats.cacheHitRateProperty());
+		this.cacheHitDregrees = cacheHitRate.multiply(-270);
+		this.cacheHitPercentage = cacheHitRate.multiply(100);
+		this.totalBytesRead = WeakBindings.bindLong(stats.toalBytesReadProperty());
+		this.totalBytesWritten = WeakBindings.bindLong(stats.toalBytesWrittenProperty());
+		this.totalBytesDecrypted = WeakBindings.bindLong(stats.totalBytesDecryptedProperty());
+		this.totalBytesEncrypted = WeakBindings.bindLong(stats.totalBytesEncryptedProperty());
+		this.filesRead = WeakBindings.bindLong(stats.filesRead());
+		this.filesWritten = WeakBindings.bindLong(stats.filesWritten());
+		this.bpsEncrypted = WeakBindings.bindLong(stats.bytesPerSecondEncryptedProperty());
+		this.bpsDecrypted = WeakBindings.bindLong(stats.bytesPerSecondDecryptedProperty());
+
+		this.ioAnimation = new Timeline(); //TODO Research better timer
+		ioAnimation.getKeyFrames().add(new KeyFrame(Duration.seconds(IO_SAMPLING_INTERVAL), new IoSamplingAnimationHandler(readData, writeData)));
+		ioAnimation.setCycleCount(Animation.INDEFINITE);
+		ioAnimation.play();
+
+		// make sure to stop animating,
+		// otherwise a global timer (GC root) will keep a strong reference to animation
+		window.setOnHiding(evt -> {
```

Comment:
Is `onHiding()` also triggered, if the window is closed or does one have to use the `onCloseRequestMethod()`?

Reply:
not sure, needs some experiments if the contract is unclear

Reply:
Seems like it is _only_ triggered by external close requests:

```
/**
  * Called when there is an external request to close this {@code Window}.
  * The installed event handler can prevent window closing by consuming the
  * received event.
  */
```

Reply:
Tested with visualvm. Works as intended.